lib/valueflow.cpp

```diff
@@ -5116,7 +5121,7 @@ static void valueFlowConditionExpressions(TokenList *tokenlist, SymbolDatabase*
             {
                 for (const Token* condTok2 : getConditions(condTok, "&&")) {
                     if (is1) {
-                        SameExpressionAnalyzer a1(condTok2, makeConditionValue(1, condTok2, true), tokenlist);
+                        SameExpressionAnalyzer a1(condTok2, makeConditionValue(1, condTok2, /*assume*/ true, /*invert*/ true), tokenlist);
```

Comment:
We should probably only set this when `condTok2` is not a comparison operator and its not bool. Similar to `is1` above but using `condTok2` instead.